IntroducingMethods/src/cst1201/IntroducingMethods.java

```diff
@@ -3,42 +3,58 @@
 import java.util.Scanner;
 
 /**
- * TODO: Description of your class.
+ * TODO: D644
  *
- * @author TODO: Your name here.
+ * @author TODO: MingJie Huang
  */
 public class IntroducingMethods {
 
     public static void main(String[] args) {
 
-        // TODO: Fill in the body with your code
     }
 
-    /**
-     * Given a Scanner, prompts the user for a String.  If the user enters an empty
-     * String, reports an error message and asks for a non-empty String.  Returns the
-     * String to the calling program.
-     * @param inScanner The Scanner to use to prompt the user.
-     * @return The resulting String from the user.
-     */
     private static String getInputString(Scanner inScanner) {
-        // TODO: Fill in the body
+        Scanner in = new Scanner(System.in);
+        System.out.print("Enter a string: ");
+        String inputString = in.nextLine();
+        if (inputString.length() > 0) {
+            System.out.println("you just entered" + inputString);
+            return inputString;
 
-        // NOTE: Do not declare a Scanner in the body of this method.
+        } else {
+            System.out.println("ERROR - string must not be empty.");
+            System.out.print("Enter a string: ");
+            return getInputString(inScanner);
+        }
     }
 
-    /**
-     * Given a String, returns the number of words in the String.  A word is a sequence of 
-     * characters with no spaces.  For example, the method call:
-     * <pre>
-     *      int count = getWordCount("The quick brown fox jumped");
-     * </pre>
-     * results in count having a value of 5. This method should be called from the main method.
-     * For this assignment, you may assume that words will be separated by exactly one space.
-     * @param input The String in question.
-     * @return The number of words in the String. Each word is separated by a single space.
-     */
     private static int getWordCount(String input) {
-        // TODO: Fill in the body
+        int count = 0;
+
+        for (int i = 0; i < input.length(); i++) {
+            if (input.charAt(i) == ' ') {
+                count++;
+            }
+        }
+        System.out.println("Amount of word that you entered:"+ count);
+        return count;
+    }
+
+    private static String getFirstWord(String input) {
+        if (input.indexOf(' ') > -1) {
+            System.out.println("First word is" +input);
```

Comment:
No output here, and also that is generally incorrect, right? The first word may not be the complete string.